include/aikido/robot/Hand.hpp

```diff
@@ -0,0 +1,51 @@
+#ifndef AIKIDO_ROBOT_HAND_HPP_
+#define AIKIDO_ROBOT_HAND_HPP_
+
+#include <set>
+#include <string>
+#include <unordered_map>
+#include <dart/dart.hpp>
+#include "aikido/control/PositionCommandExecutor.hpp"
+#include "aikido/io/yaml.hpp"
+#include "aikido/robot/GrabMetadata.hpp"
+
+namespace aikido {
+namespace robot {
+
+AIKIDO_DECLARE_POINTERS(Hand)
+
+/// Abstract class for Hand interface.
+class Hand
+{
+public:
+  virtual ~Hand() = default;
+
+  /// Grabs an object. Immediately executes.
+  /// \param[in] bodyToGrab The object to grab
+  virtual void grab(const dart::dynamics::SkeletonPtr& bodyToGrab) = 0;
+
+  /// Ungrabs an object. Immediately ungrabs.
+  /// Throws a runtime_error if fails.
```

Comment:
Doxygen has a [`\throws`](https://www.stack.nl/~dimitri/doxygen/manual/commands.html#cmdthrows) command that we can use on these methods.